coveo/ui-kit
# fix(headless): save facet order in the history state

This PR saves the facet order in the state and introduces a `HistoryState` interface to contain new values that aren't part of a response.

If you wish to test this PR, you can do the following:
1. Search “australia” (facet order should be “File sizes” then “Authors”)
2. Filter on “BBC News” (facet order should be “File sizes” then “Authors”)
3. Search “country” (facet order should be “Authors” then “File sizes”)
4. Press the “BACK” history button (facet order should be “File sizes” then “Authors”)
5. Press the "FORWARD" history button (facet order should be “Authors” then “File sizes”)
